R/numbered_checks.R

```diff
@@ -334,4 +334,28 @@ check_openclos_iandd <- function(form, rep){
              (openclos == 1 & iandd == 3) |
              (openclos == 2 & (iandd == 1 | iandd == 2)))
   return(problems)
+}
+
+#' Check that he time difference between the time from injury to hsp admission and 
+#' the time from injury to prep solution in ER is within +/- 24 hr range
+#' 
+#' @param form a dataframe containing form1.1, form4.1, form5.14
+#' @return a dataframe containing problematic entries
+#' 
+#' @import tidyverse
+#' @export
+check_ihunits_ipunits <- function(form){
+  problems <- form %>% transmute(
+    region, site, studyid, ptinit, ptstatus, 
+    ihunits, ihdays, ihhrs, ipunits1, ipreph1, iprepd1, ipunits2, ipreph2, iprepd2,
+    ipunits3, ipreph3, iprepd3,
+    comment = "The time difference between the time from injury to hsp admission and the time from injury to prep solution in ER should be within +/- 24 hr range") %>% 
+    filter(ptstatus == 1 & 
+             ((ihunits == 1 & ipunits1 == 1 & abs(ihhrs - ipreph1) >= 24) | 
```

Comment:
Repetitively checking ihunits == 1 is inefficient. Use brackets to check them only once (ihunits == 1 & (the rest))